clang/lib/Format/TokenAnnotator.cpp

```diff
@@ -6157,6 +6157,12 @@ bool TokenAnnotator::canBreakBefore(const AnnotatedLine &Line,
     return !(Previous && (Previous->is(tok::kw_for) || Previous->isIf()));
   }
 
+  if (Left.isOneOf(tok::r_paren, TT_TrailingAnnotation) &&
```

Comment:
Is this related?

Reply:
yes, it is part of the problems reported in #55731